sdk/common.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2024 Edgeless Systems GmbH
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package sdk
+
+import (
+	"fmt"
+	"log/slog"
+	"os"
+	"path/filepath"
+
+	"github.com/edgelesssys/contrast/internal/atls"
+	"github.com/edgelesssys/contrast/internal/attestation/certcache"
+	"github.com/edgelesssys/contrast/internal/attestation/snp"
+	"github.com/edgelesssys/contrast/internal/attestation/tdx"
+	"github.com/edgelesssys/contrast/internal/fsstore"
+	"github.com/edgelesssys/contrast/internal/logger"
+	"github.com/edgelesssys/contrast/internal/manifest"
+)
+
+const cacheDirEnv = "CONTRAST_CACHE_DIR"
+
+// ValidatorsFromManifest returns a list of validators corresponding to the reference values in the given manifest.
+// Originally an unexported function in the contrast CLI.
+// Can be made unexported again, if we decide to move all userapi calls from the CLI to the SDK.
+func ValidatorsFromManifest(m *manifest.Manifest, log *slog.Logger, hostData []byte) ([]atls.Validator, error) {
```

Comment:
I have mixed feelings about debug logging in a library. Should it at least be self-configuring?

Reply:
Why so? I often hear this sentiment but haven't gotten around to asking where it comes from. 
Self-configuring means that there is no logger argument and the function instead configures loggers transparently?

I intentionally did not change the function because it could be an SDK-internal function if the gRPC calls from `contrast set` and `contrast recover` were also in the SDK. Not exporting this function would be the better solution imo. However, I did not want to move those two calls into SDK pkg to keep the PR smaller.

Reply:
Mostly because you need to propagate logger structs through your app, even if you did not mean to add logging to begin with. You can also argue that a library should communicate through its arguments and returns, without side effects. This also goes for the cache dir: the `go-sev-guest` library did the right thing by accepting an `HttpsGetter` interface - then the side effects are with the caller.

Otoh I've been annoyed more than once by a library that did not expose logging but logged anyway in a different style.

We could also do a `New()` and `NewWithSLog(*slog.Logger)`.

Reply:
> Mostly because you need to propagate logger structs through your app, even if you did not mean to add logging to begin with.

I'd very much want the option to configure logging for libraries. If the consumer doesn't want it, they can pass a logger that discards the output. Related discussion: https://go-review.googlesource.com/c/go/+/626486 https://github.com/golang/go/issues/62005